### PR TITLE
Update PlayerPostRespawnEvent to check for anchor respawn

### PR DESCRIPTION
**Pull request by MisterVector**
This PR adds a new method to the PlayerPostRespawnEvent event, isAnchorSpawn(), to check if the respawn was due to an anchor.

### CODE REVIEW

**Review by Machine-Maker — CHANGES_REQUESTED** [review]
Should probably be just added to the patch that adds this event seeing as it’s a Paper-specific event.

**Comment by MisterVector** [comment]
You're right about that. Will do!

**Comment by MisterVector** [comment]
On second thought I'm not sure of the implications of doing that, as the variable I'm trying to pass as an argument did not exist at the time of that patch's creation. What would be the best approach if this was going to be accomplished?

**Comment by Machine-Maker** [comment]
> What would be the best approach if this was going to be accomplished?

You can either re-order the patches, as in find the patch that adds the variable that you need, get its number, add 1 to it, and rename the patch file to that. (make sure you re-order both the server and api patches to they are still "in sync") 

-- OR --

You can add whatever you need to the patch you are editing, and then remove it from the patch that comes after that also adds it.

**Comment by MisterVector** [comment]
Rebased both api and server patches with the specified changes. Hopefully this won't be hard to sort out before merging.

I did add `isAnchorSpawn = flag3; // Paper - Fix anchor respawn acting as a bed respawn from the end portal` which is kind of strange given the comment not being within the context of this PR. Anyways, things should hopefully be ok from my end.

**Comment by MisterVector** [comment]
On second thought I shouldn't have added `PlayerRespawnEvent respawnEvent = new PlayerRespawnEvent(respawnPlayer, location, isBedSpawn && !isAnchorSpawn, isAnchorSpawn); // Paper - Fix anchor respawn acting as a bed respawn from the end portal` to my patch, the assignment of isAnchorSpawn should have been enough as I included it in my event call.

**Comment by MisterVector** [comment]
I will redo this again, trying to leave out the things that should not be included.

File: patches/server/0319-Implement-PlayerPostRespawnEvent.patch [hunk 1]
```diff
@@ -24,22 +25,30 @@ index abba62b7252c6d6ee92a4b1a9b23df7758ab6a77..7dd874c6aa1268665d4a64dfc41013c3
              ServerLevel worldserver1 = this.server.getLevel(entityplayer.getRespawnDimension());
              if (worldserver1 != null) {
                  Optional optional;
-@@ -776,6 +781,7 @@ public abstract class PlayerList {
+@@ -739,6 +745,7 @@ public abstract class PlayerList {
+                 if (optional.isPresent()) {
+                     BlockState iblockdata = worldserver1.getBlockState(blockposition);
+                     boolean flag3 = iblockdata.is(Blocks.RESPAWN_ANCHOR);
++                    isAnchorSpawn = flag3;
```
Comment (NoahvdAa) [inline patches/server/0319-Implement-PlayerPostRespawnEvent.patch:32]: Missing Paper comment.
```suggestion
+                    isAnchorSpawn = flag3; // Paper
```

**Review by NoahvdAa — APPROVED** [review]

**Comment by NoahvdAa** [comment]
Isn't this a duplicate of #5465?

**Comment by MisterVector** [comment]
Yep... did not look for this beforehand, wasn't actually expecting someone wanted to do this before... well I'll close this out then.

**Pull request closed without merging**